src/bin/ion/commands/generate/generator.rs

```diff
@@ -319,11 +489,20 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
         code_gen_context: &mut CodeGenContext,
     ) -> CodeGenResult<()> {
         // Add namespace to tera context
-        if let Some(namespace) = self.namespace {
-            context.insert("namespace", namespace);
+        let mut import_context = Context::new();
+        if let Some(ref mut namespace) = self.current_type_fully_qualified_name {
+            let namespace_ref = namespace.as_slice();
+            context.insert("namespace", &namespace_ref[0..namespace_ref.len() - 1]);
+            import_context.insert("namespace", &namespace_ref[0..namespace_ref.len() - 1]);
         }
         // Render or generate file for the template with the given context
-        let template: &Template = &code_gen_context.abstract_data_type.as_ref().try_into()?;
+        let template: &Template = &code_gen_context.data_model_node.as_ref().try_into()?;
+
+        // Since `java` templates use recursion(i.e. use the same template for nested types) when rendering nested types,
+        // We need to tune the `is_nested` flag to allow static classes being added inside a parent class
+        if L::name() == "java" {
+            context.insert("is_nested", &false);
+        }
```

Comment:
This is not specific to Java but since Rust support is disabled for Rust in this PR its just used for Java in this PR. Although I will remove all the references like these where its only for one programming language and ignored for another. i.e. Even though the other language doesn't use it, we can just add it for that language and ignore it.